chain/cmd/bandsv/main.go

```diff
@@ -70,42 +78,126 @@ type Proof struct {
 	BlockRelayProof BlockRelayProof `json:"blockRelayProof"`
 }
 
-func main() {
-	r := gin.Default()
+const priv = "06be35b56b048c5a6810a47e2ef612eaed735ccb0d7ea4fc409f23f1d1a16e0b"
+const port = "5001"
+const nodeURI = "http://localhost:26657"
+const queryURI = "http://localhost:1317"
+
+var rpcClient *rpc.HTTP
+var pk secp256k1.PrivKeySecp256k1
 
-	r.POST("/request", func(c *gin.Context) {
-		var json OracleRequest
-		if err := c.ShouldBindJSON(&json); err != nil {
-			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+// TODO
+// - Add query from rest client and ask via that endpoint
+func HasCode(codeHash []byte) (bool, error) {
+	key := zoracle.CodeHashStoreKey(codeHash)
+	resp, err := rpcClient.ABCIQuery("/store/zoracle/key", key)
+	if err != nil {
+		return false, err
+	}
+
+	return len(resp.Response.Value) > 0, nil
+}
+
+func handleRequestData(c *gin.Context) {
+	var requestData OracleRequest
+	if err := c.ShouldBindJSON(&requestData); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	if len(requestData.CodeHash) == 0 && len(requestData.Code) == 0 {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Missing code/codeHash"})
+		return
+	}
+	if len(requestData.CodeHash) > 0 && len(requestData.Code) > 0 {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Only one of code/codeHash can be sent"})
+		return
+	}
+
+	// TODO
+	// Need some work around to make params can be empty bytes
+	if len(requestData.Params) <= 0 {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Params should not be empty bytes"})
+		return
+	}
+
+	if len(requestData.Code) > 0 {
+		requestData.CodeHash = crypto.Sha256(requestData.Code)
```

Comment:
Use getCodeHash from x/zoracle from this [file](https://github.com/bandprotocol/d3n/blob/master/chain/x/zoracle/internal/types/code.go)